### ISSUE TITLE
Per-file configuration is applied to all files

### ISSUE THREAD

**Issue by kaiyoma** [lesshint/lesshint#388]
`package.json`:

```{
  "name": "test",
  "scripts": {
    "lesshint": "lesshint styles-1.less styles-2.less"
  },
  "devDependencies": {
    "lesshint": "3.3.1"
  }
}
```

`styles-1.less`:

```
.test1 {
  color:blue;
}
```

`styles-2.less`:

```
.test2 {
  color:red;
}
```

`npm run lesshint` results in two warnings, as expected:

```
Warning: styles-1.less: line 2, col 9, spaceAfterPropertyColon: Colon after property name should be followed by one space.
Warning: styles-2.less: line 2, col 9, spaceAfterPropertyColon: Colon after property name should be followed by one space.
```

Now let's add an ignore directive to only `styles-1.less`, so it becomes:

```
// lesshint spaceAfterPropertyColon: false
.test1 {
  color:blue;
}
```

I would expect that the warning from the first file would go away, but we'd still get the warning from the second file. However, that's not the case; both warnings go away, presumably because once the ignore directive is processed, it stays valid until told otherwise. This is not intuitive at all and doesn't match up with other linters like eslint. It's expected that a per-file configuration is valid for only that file.

**Comment by jwilsson** [comment]
Confirmed, this is a bug.

I think I know what's causing it though, I'll push a fix shortly.

**Comment by jwilsson** [comment]
This should be fixed by da93ccb8359df9602c64afcd228dd97105a05589.
